Composer/packages/extension-client/src/hooks/useMenuConfig.ts

```diff
@@ -8,7 +8,20 @@ import { EditorExtensionContext } from '../EditorExtensionContext';
 import { MenuUISchema } from '../types';
 
 export function useMenuConfig(): MenuUISchema {
-  const { plugins } = useContext(EditorExtensionContext);
+  const { plugins, shellData } = useContext(EditorExtensionContext);
+  const uiSchema = plugins.uiSchema || {};
+  const sdkSchema = shellData.schemas?.sdk;
+  const sdkDefinitions = sdkSchema?.content?.definitions || {};
 
-  return useMemo(() => mapValues(plugins.uiSchema, 'menu'), [plugins.uiSchema]);
+  return useMemo(() => {
+    const menuSchema = mapValues(uiSchema, 'menu') as MenuUISchema;
+    const implementedMenuSchema = Object.entries(menuSchema).reduce((result, [$kind, menuOpt]) => {
```

Comment:
Do you need the reduce here? Could this be a map that is setting value in an associative array?

Reply:
Yeah, I made it obscure, that will be a more readable way.